### PR TITLE
Simplebot TGUI conversion

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Creates a modular TGUI for simple bots.

**Current issues with simplebots:**
- They are all HTML

**Current pictures:** 
Medbot interface
Imgur img
Beepsky
Imgur img
Floorbot, emagged by borg
Imgur img

### TODO
- [x] Create the UI
- [x] Convert bots over to opening it
- [x] Create parent proc for UI data
- [x]  Fix authZ checks on UI actions
- [x] UI - Settings section
- [x] UI - Hack button for bad sillycons
- [x] UI - PAI information
- [x] Ensure mulebot still callable
- [ ] Fix cleanbot disappearing on power on // this is still an issue but I don't think it's related
- [x] Bug fixes, surely
- [x] Test

## Why It's Good For The Game
- Better looking UI
- Many new tooltips for information
- You can now lock/unlock the controls via UI
- You can now lock/unlock the maintenance hatch via UI
- A little bit easier to catch runaway bots

## Changelog

:cl:
add: All of your favorite simple bot friends aboard the station now have a much better interface in TGUI
qol: You can now lock and unlock bots, including their maintenance hatches, via the UI.
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
LET'S FUCKING GOOOOOOOOOOOOOOOOOOOOOO

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 1]
```diff
@@ -349,54 +349,54 @@
...
-		dat += "<A href='byond://?src=[REF(src)];op=sethome'>Set Home</A><BR>"
-		dat += "<A href='byond://?src=[REF(src)];op=autoret'>Toggle Auto Return Home</A> ([auto_return ? "On":"Off"])<BR>"
-		dat += "<A href='byond://?src=[REF(src)];op=autopick'>Toggle Auto Pickup Crate</A> ([auto_pickup ? "On":"Off"])<BR>"
-		dat += "<A href='byond://?src=[REF(src)];op=report'>Toggle Delivery Reporting</A> ([report_delivery ? "On" : "Off"])<BR>"
-		if(load)
-			dat += "<A href='byond://?src=[REF(src)];op=unload'>Unload Now</A><BR>"
-		dat += "<div class='notice'>The maintenance hatch is closed.</div>"
-
-	return dat
+// /mob/living/simple_animal/bot/mulebot/get_controls(mob/user)
```
Comment (stylemistake) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:352]: remove commented code

Comment (jlsnow301) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm:352]: I'm a bit worried about this- I think inside mulebot it said it /needed/ these in order for pda control to work

**Comment by jlsnow301** [comment]
just to think out loud: using large buttons isn't standard practice for tgui but it had a nicer look and feel this way. i'd really hate to go over this only to use ```<Labeledlist>``` over the controls. the result would be a very plain "control: button" section

**Comment by tralezab** [comment]
> just to think out loud: using large buttons isn't standard practice for tgui but it had a nicer look and feel this way. i'd really hate to go over this only to use `<Labeledlist>` over the controls. the result would be a very plain "control: button" section

If you can make it look good, then it's cool. There's a lot of uis that break normal conventions and look all the better for it.

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 1 above)
Comment (JohnFulpWillard) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm:352]: Mulebot snowflakes a ton of code and it even has its own UI, so im unsurprised.
You can look at ``/obj/item/cartridge/Topic(href, href_list)`` and ``/obj/item/cartridge/proc/bot_control()`` to see how PDA's deal with them, they have snowflake code compared to every other bot.

**Comment by Ziiro** [comment]
Request: Can we use the different value slider (the vertical bar? I forget the name) for the health threshold? Having that as a knob seems weird.

**Comment by jlsnow301** [comment]
> Request: Can we use the different value slider (the vertical bar? I forget the name) for the health threshold? Having that as a knob seems weird.
> 
> Sorry to bikeshed, this otherwise is fantastic.

no worries, i'll take a look. i am also not very happy with the knob

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 1 above)
Comment (jlsnow301) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm:352]: Might need to create a new remote controls proc for these bots. I axed get_controls() to turn it into ui_data(), haven't looked to see if ALL bot control is dependent on it. If I even look at pda code this is going to become a pda refactor too

**Pull request merged**